op-program/verify/verify.go

```diff
@@ -202,7 +203,7 @@ func (r *Runner) run(ctx context.Context, l1Head common.Hash, agreedBlockInfo et
 	}
 	fmt.Printf("Configuration: %s\n", argsStr)
 
-	if runInProcess {
+	if r.runInProcess {
```

Comment:
If not in-process, `op-prorgram-host` needs to accept either network name `--network` or a rollup config file.  However, for the devnet, we do not have a network name for it (e.g., "op-sepolia"), and thus supplying an external rollup config file is a must.  One simple solution may accept a network name by id (e.g., `--network 900`) here https://github.com/ethstorage/optimism/blob/7ab9ae144b033e971b103bd756e08f456c856b52/op-program/host/config/config.go#L179. What do you think?

Reply:
oh, interesting yeah I hadn't thought of that.  Given op-challenger needs to run it in a separate process we definitely need a way to do it.  I like the idea of the `--network` flag accepting either a chain name or a chain ID.  The chain ID _should_ be all we use from the network name anyway, though I see we're passing the config through the host which isn't ideal.

I've put up https://github.com/ethereum-optimism/optimism/pull/12420 to avoid returning the configs via the preimage oracle unless we're using the custom chain ID marker just to be extra safe. Otherwise I think the change to op-program is straight forward if we load the config via this new process as you've done here in `verify.go`.

It would kind of be nice if op-program host didn't load the rollup config and l2 chain config unless it had to provide it via the preimage oracle but that's a fairly big change so definitely not something to mix into this PR and maybe not worth doing at all.